### PR TITLE
add exception classes and AzureCredentialWrapper

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java [hunk 1]
```diff
@@ -28,24 +30,30 @@ public static AuthType parseAuthType(String type) {
...
-            case "managed_identity":
-                return MANAGED_IDENTITY;
             case "oauth2":
                 return OAUTH2;
             case "visual_studio":
                 return VISUAL_STUDIO;
-            case "auto":
-                return AUTO;
+            default:
+                throw new UnsupportedOperationException(String.format("Invalid auth type '%s', supported values are: %s.", type,
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java:54]: use `IllegalArgumentException` here?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/exception/AzureLoginException.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.exception;
+
+public class AzureLoginException extends Exception {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/exception/AzureLoginException.java:9]: Do we need to have a parent class like `AzureException`

Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/exception/AzureLoginException.java:9]: this is auth lib, all auth exceptions should inherit this exception

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java (hunk 1 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java]: InvalidConfigurationException  will be better

**Review by Flanker32 — APPROVED** [review]
LGTM~

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/exception/LoginFailureException.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+package com.microsoft.azure.tools.auth.exception;
+
+public class LoginFailureException extends AzureLoginException {
+    private static final long serialVersionUID = -94207672112213624L;
+
+    public LoginFailureException(String message) {
+        super(message);
+    }
+
+}
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/exception/LoginFailureException.java:16]: Is it really necessary for us to create these new exceptions whose only difference is the class name?
how about only create one/two for us, let us use different values of its properties to describe different cases?

Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/exception/LoginFailureException.java:16]: Rule 2:
image

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java [hunk 4]
```diff
@@ -8,44 +8,53 @@
...
-            case "managed_identity":
-                return MANAGED_IDENTITY;
             case "oauth2":
                 return OAUTH2;
             case "visual_studio":
                 return VISUAL_STUDIO;
-            case "auto":
-                return AUTO;
+            default:
+                throw new InvalidConfigurationException(String.format("Invalid auth type '%s', supported values are: %s.", type,
```
[the remark this answers is not in the document]
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java:55]: This piece of code is running without costing too much memory and it merely runs, using a map(hash map) will cost permanent memory in heap.

**Pull request merged**